pkg/detector/detector.go

```diff
@@ -1347,8 +1325,46 @@ func (d *ResourceDetector) HandleClusterPropagationPolicyCreationOrUpdate(policy
 	return nil
 }
 
-// CleanupLabels removes labels from object referencing by objRef.
-func (d *ResourceDetector) CleanupLabels(objRef workv1alpha2.ObjectReference, labelKeys ...string) error {
+func (d *ResourceDetector) cleanupPolicyRefNamespaceScopeResources(policyNS, policyName string, rb *workv1alpha2.ResourceBinding, removeLabels RemoveMapKeysFunc) error {
+	// Must remove the labels from the resource template ahead of ResourceBinding, otherwise might lose the chance
+	// to do that in a retry loop(in particular, the label was successfully removed from ResourceBinding, but
+	// resource template not), since the ResourceBinding will not be listed again.
+	if err := d.cleanupResourceTemplateLabels(rb.Spec.Resource, removeLabels); err != nil {
+		klog.Errorf("Failed to clean up label from resource(%s-%s/%s) when propagation policy(%s/%s) removing, error: %v",
+			rb.Spec.Resource.Kind, rb.Spec.Resource.Namespace, rb.Spec.Resource.Name, policyNS, policyName, err)
+		return err
```

Comment:
The function `cleanupPolicyRefNamespaceScopeResources` is inconsistent with the original implementation. When `cleanupResourceTemplateLabels` fails, it will directly throw an error and return. However, the original implementation will continue to execute `cleanupRBLabels`.